src/mscorlib/shared/System/StringComparer.cs

```diff
@@ -244,17 +323,22 @@ public override int GetHashCode(string obj)
                 throw new ArgumentNullException(nameof(obj));
 #endif
             }
-            return obj.GetHashCode();
+            return TextInfo.GetHashCodeOrdinalIgnoreCase(obj);
         }
 
         // Equals/GetHashCode methods for the comparer itself. 
-        public override bool Equals(object obj) => obj is OrdinalComparer;
-        public override int GetHashCode() => nameof(OrdinalComparer).GetHashCode();
+        public override bool Equals(object obj) => obj is OrdinalCaseSensitiveComparer;
```

Comment:
> increasing memory footprint because of the _ignoreCase field

This does not matter. These are singletons. And the extra field will actually fit into the default padding that each object has, so it won't actually increase the memory footprint at all.

> Do we need the inheritance just because of the Equality check?

It is the easiest way to make the Equality check work.